src/site/content/en/angular/virtualize-lists-with-cdk/index.md

```diff
@@ -0,0 +1,118 @@
+---
+layout: post
+title: Virtualize large lists with the Angular CDK
+subhead: Make large lists more responsive by implementing virtual scrolling.
+authors:
+  - sfluin
+date: 2019-07-12
+# Add an updated date to your post if you edit in the future.
+# updated: 2019-06-27
+hero: hero.jpg
+# You can adjust the position of your hero image with this property.
+# Values: top | bottom | center (default)
+# hero_position: bottom
+alt: Stacks of vinyl records.
+description: |
+  Learn how to make large lists more responsive by implementing virtual scrolling with the Angular Component Dev Kit.
+tags:
+  - angular
+---
+
+The scrolling list is one of the most common UI patterns today, whether it's browsing an infinitely scrolling feed on your favorite social media site, or navigating an enterprise dashboard. When scrolling lists become very long (hundreds, thousands, or hundreds of thousands of items), application performance can suffer.
+
+Large lists can be slow to load because the application must load and render all the data up front. They can also be slow to render and navigate because each item in the list can have rich data , media, and functionality.
+
+Users can experience problems when they load or scroll the page, leading to frustration and page abandonment.
+
+## Virtual scrolling in Angular with the Component Dev Kit
+Virtual scrolling is the primary technique used to address these scale problems. Virtual scrolling gives the impression of a very large list—by providing an appropriately sized scroll bar—and the ability to navigate the list without requiring the application to hold the entire list in memory or render it on the page.
+
+In Angular, virtual scrolling is provided by the [Component Dev Kit (CDK)](https://material.angular.io/cdk/categories). By modifying the way you iterate through lists, and by supplying a couple of additional configuration parameters, the CDK's virtual scrolling will automatically manage the virtual rendering of your lists, improving page performance and responsiveness.
+
+Instead of rendering the entire list at a time, only a subset of the items that fits on the screen (plus a small buffer) will be rendered. As the user navigates, a new subset of items is calculated and rendered, re-using the existing DOM if desired.
+
+The rest of this post walks through how to set up basic virtual scrolling. You can see a [full working example on StackBlitz](https://stackblitz.com/edit/scroll-list?file=src/app/app.component.ts).
```

Comment:
Do you think we can directly embed the example? We already do that in 2 other posts.

Reply:
That works